exercises/pig-latin/src/example.c

```diff
@@ -0,0 +1,121 @@
+#include "pig_latin.h"
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <ctype.h>
+#include <stdbool.h>
+
+static bool is_vowel(const char a)
+{
+   return a == 'a' || a == 'e' || a == 'i' || a == 'o' || a == 'u';
+}
+
+static bool starts_with_vowel(const char a, const char b)
+{
+   return is_vowel(a) || (a == 'x' && b == 'r') || (a == 'y' && b == 't');
+}
+
+static void count_words_and_chars(const char *input, size_t * words,
```

Comment:
I'd say split this into `count_words()` and `count_chars()` - that would align better with the single responsibility principle and would make the parameters simpler. True, it would result in two iterations through the input but we're not expecting to handle massive strings.

Reply:
The whole point of this function was to spare one iteration through the input. If possible, I would prefer to keep it this way.